formats/json-jackson/src/main/java/io/cloudevents/jackson/JsonFormat.java

```diff
@@ -47,41 +47,95 @@
     private final ObjectMapper mapper;
     private final boolean forceDataBase64Serialization;
     private final boolean forceStringSerialization;
+    private final boolean forceExtensionNameLowerCaseDeserialization;
+    private final boolean forceIgnoreInvalidExtensionNameDeserialization;
 
     /**
      * Create a new instance of this class customizing the serialization configuration.
      *
      * @param forceDataBase64Serialization force json base64 encoding for data
      * @param forceStringSerialization     force string serialization for non json data field
+     * @param forceExtensionNameLowerCaseDeserialization        force extension name deserialization for lower case
+     * @param forceIgnoreInvalidExtensionNameDeserialization    force extension name deserialization for ignoring invalid name
      * @see #withForceJsonDataToBase64()
      * @see #withForceNonJsonDataToString()
+     * @see #withForceExtensionNameLowerCaseDeserialization()
+     * @see #withForceIgnoreInvalidExtensionNameDeserialization()
      */
-    public JsonFormat(boolean forceDataBase64Serialization, boolean forceStringSerialization) {
+    public JsonFormat(
+        boolean forceDataBase64Serialization,
+        boolean forceStringSerialization,
+        boolean forceExtensionNameLowerCaseDeserialization,
+        boolean forceIgnoreInvalidExtensionNameDeserialization
+    ) {
```

Comment:
Can we create a class that holds these configurations `forceDataBase64Serialization`, `forceStringSerialization`, etc rather than having a lot of boolean parameters?

Also, if we create a class we can create a different constructor so that we're not making breaking changes.

Reply:
Maybe we can have a builder class with a fluent interface to build such configuration objects.

Reply:
@pierDipi 
Create JsonFormatOptions class with builder.
It has serialization / deserialization boolean options.
And it does not breaking change